frontend/reducers/index.js

```diff
@@ -1,25 +1,50 @@
-import {FETCH_RESULT, SHOW_MAP} from './../components/ButtonField/ButtonField.Action';
+import {
+    FETCH_RESULT,
+    SHOW_MAP,
+    ADD_DATA,
+} from './../components/ButtonField/ButtonField.Action';
 
 const initialState = {
     data: [],
     currentInformation: {
-        locationX: 10.11,
-        locationY: 12.13,
+        location: {
+            x: 10.11,
+            y: 12.13,
+        },
         signal: 100,
-        provider: 'Undefinied',
+        provider: 'Telekom',
+        connectionType: '4G',
+        platform: 'Android', // test data for now / graphql expects Android/IOs
     },
 };
 
 export default function reducer(state = initialState, action) {
     switch (action.type) {
         case FETCH_RESULT:
-            return { ...state, array: action.payload.result };
+            return { ...state, data: action.payload.result };
+        case ADD_DATA:
+            return addDataConnectionReducer(state, action);
         case SHOW_MAP:
             return {
                 ...state,
-                showMap: action.payload
-            }
+                showMap: action.payload,
+            };
         default:
             return state;
     }
 }
+
+function addDataConnectionReducer(state, { payload }) {
+    const elements = payload.dataConnectionInformation;
+
+    return {
+        ...state,
+        data: addElementToArray(elements, state.data),
```

Comment:
Warning `[Unhandled promise rejection: TypeError: Invalid attempt to spread non-iterable instance]`